codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/StructureGenerator.kt

```diff
@@ -50,6 +50,13 @@ fun redactIfNecessary(member: MemberShape, model: Model, safeToPrint: String): S
     }
 }
 
+/**
+ * The artifact type for whom we are generating the structure.
+ */
+public enum class TYPE {
```

Comment:
> In general, enum classes shouldn't be named with all caps. Thinking about this name though, I'm concerned Type will be too easy to confuse with the other various representations of type in the codegen. Perhaps we should call this Target or CodegenTarget?

Agree, I was hoping someone could come up with a better name. I've renamed to `CodegenTarget`. https://github.com/awslabs/smithy-rs/pull/1174/commits/c63b52efaaa450c931a51d818311d6a6b449110c

> It's probably useful to have outside of StructureGenerator so it can be used in other places, but we can always move it when using it again.

The classpath looks meaningful. Other places are importing it already via:

```
import software.amazon.smithy.rust.codegen.smithy.generators.CodegenTarget
```

You're right though in that the _file_ where the `enum` is placed might not be intuitive. I didn't want to create a `Utils.kt` file given that we've managed without one so far, but I couldn't think of any other more appropriate file.